pkg/apis/core/validation/events_test.go

```diff
@@ -1312,3 +1312,73 @@ func TestValidateEventUpdateForNewV1Events(t *testing.T) {
 		}
 	}
 }
+
+func TestEventV1EventTimeImmutability(t *testing.T) {
+	testcases := []struct {
+		Name  string
+		Old   metav1.MicroTime
+		New   metav1.MicroTime
+		Valid bool
+	}{
+		{
+			Name:  "noop microsecond",
+			Old:   metav1.NewMicroTime(time.Unix(100, int64(5*time.Microsecond))),
+			New:   metav1.NewMicroTime(time.Unix(100, int64(5*time.Microsecond))),
+			Valid: true,
+		},
+		{
+			Name:  "noop nanosecond",
+			Old:   metav1.NewMicroTime(time.Unix(100, int64(5*time.Nanosecond))),
+			New:   metav1.NewMicroTime(time.Unix(100, int64(5*time.Nanosecond))),
+			Valid: true,
+		},
+		{
+			Name:  "change microseconds down",
+			Old:   metav1.NewMicroTime(time.Unix(100, int64(5*time.Microsecond))),
+			New:   metav1.NewMicroTime(time.Unix(100, int64(5*time.Microsecond-time.Nanosecond))),
+			Valid: false,
+		},
+		{
+			Name:  "change microseconds up",
+			Old:   metav1.NewMicroTime(time.Unix(100, int64(5*time.Microsecond-time.Nanosecond))),
+			New:   metav1.NewMicroTime(time.Unix(100, int64(5*time.Microsecond))),
+			Valid: false,
+		},
+		{
+			Name:  "change nanoseconds down",
```

Comment:
slightly misleading title -- you can't increment/decrement over a microsecond boundary...

Reply:
fixed testcase names